Simplified/Reader2/Internal/ReaderModule.swift

```diff
@@ -14,120 +15,78 @@ import UIKit
-/// The ReaderModule handles the presentation of publications to be read by the
-/// user. It contains sub-modules implementing ReaderFormatModule to handle each
-/// format of publication (eg. CBZ, EPUB).
+/// The ReaderModule handles the presentation of publications to be read by the user.
+/// It contains sub-modules implementing ReaderFormatModule to handle each format of publication (eg. CBZ, EPUB).
 protocol ReaderModuleAPI {
-
-  var delegate: ReaderModuleDelegate? { get }
-
-
+  
+  var delegate: ModuleDelegate? { get }
+  
   /// Presents the given publication to the user, inside the given navigation controller.
-  ///
   /// - Parameter publication: The R2 publication to display.
   /// - Parameter book: Our internal book model related to the `publication`.
-  /// - Parameter navigationController: The navigation stack the book will be
-  /// presented in.
+  /// - Parameter navigationController: The navigation stack the book will be presented in.
   /// - Parameter completion: Called once the publication is presented, or if an error occured.
-  func presentPublication(_ publication: Publication,
-                          book: NYPLBook,
-                          in navigationController: UINavigationController,
-                          completion: @escaping () -> Void)
-
-}
-
-protocol ReaderModuleDelegate: ModuleDelegate {
-
-  /// Called when the reader needs to load the R2 DRM object for the given publication.
-  func readerLoadDRM(for book: NYPLBook,
-                     completion: @escaping (CancellableResult<DRM?>) -> Void)
-
+  func presentPublication(publication: Publication, book: NYPLBook, in navigationController: UINavigationController, completion: @escaping () -> Void)
+  
 }
 
-
 final class ReaderModule: ReaderModuleAPI {
-
-  weak var delegate: ReaderModuleDelegate?
+  
+  weak var delegate: ModuleDelegate?
   private let resourcesServer: ResourcesServer
-
+  
   /// Sub-modules to handle different publication formats (eg. EPUB, CBZ)
   var formatModules: [ReaderFormatModule] = []
-
-  init(delegate: ReaderModuleDelegate, resourcesServer: ResourcesServer) {
+  
+  init(delegate: ModuleDelegate?, resourcesServer: ResourcesServer) {
     self.delegate = delegate
     self.resourcesServer = resourcesServer
-
+    
     formatModules = [
-      EPUBModule(delegate: self),
+      EPUBModule(delegate: self)
     ]
 
-    // TODO: do we need to handle PDFs in R2?
-    //        if #available(iOS 11.0, *) {
-    //            formatModules.append(PDFModule(delegate: self))
-    //        }
   }
-
-  func presentPublication(_ publication: Publication,
+  
+  func presentPublication(publication: Publication,
                           book: NYPLBook,
                           in navigationController: UINavigationController,
                           completion: @escaping () -> Void) {
     guard let delegate = delegate else {
-      preconditionFailure("Reader delegate not set")
+      fatalError("Reader delegate not set")
```

Comment:
I don't think we should crash here, since the delegate is not even that essential (it just presents errors). I would just log an error.

Reply:
we probably shouldn't have this situation in first place, NYPLR2Owner is that delegate and there's no situation when we initialize ReaderModule without it. Made this property not optional.